src/quicklist.c

```diff
@@ -2008,6 +2008,32 @@ int quicklistTest(int argc, char *argv[], int accurate) {
             quicklistRelease(ql);
         }
 
+        TEST("insert head while head node is full") {
+            quicklist *ql = quicklistNew(4, options[_i]);
+            for (int i = 0; i < 10; i++)
+                quicklistPushTail(ql, genstr("hello", i), 6);
+            quicklistSetFill(ql, -1);
+            quicklistEntry entry;
+            quicklistIndex(ql, 0, &entry);
+            char buf[4100] = {0};
+            quicklistInsertBefore(ql, &entry, &buf, 4096);
+            ql_verify(ql, 4, 11, 1, 2);
```

Comment:
It would be better to add an assert here, to determine if the old pointer has changed to determine if it has been merged.

Reply:
since the lager ziplist will be used as merge target, so the old_head/tail will be del, we assert old_head/tail == new_head/tail？

Reply:
Sorry, I tested it and found that assert is not necessary, because ```ql_verify``` already verifies that it has been merged or not.
The code you last commit will compile error.